### ISSUE TITLE
Query by link(s)

### ISSUE THREAD

**Issue by Viper-Bit** [isar/isar#440]
**Steps to Reproduce**
I cant query by link(s), my query by link(s) every time returns empty list
```
flutter: Class of John`s teacher: Math
flutter: Founded students with `Math` class: 0
```

**Code sample**
```dart
@Collection()
class Teacher {
  int id = Isar.autoIncrement;

  @Backlink(to: 'teacher')
  final students = IsarLinks<Student>();

  late String subject;
}

@Collection()
class Student {
  int id = Isar.autoIncrement;

  late String name;

  final teacher = IsarLink<Teacher>();
}

  var teacher = Teacher()..subject = 'Math';
  var teacher2 = Teacher()..subject = 'Algebra';

  await isar
      .writeTxn(() async => await isar.teachers.putAll([teacher, teacher2]));

  var st = Student()
    ..name = 'John'
    ..teacher.value = teacher;

  var st2 = Student()
    ..name = 'Jack'
    ..teacher.value = teacher;

  var st3 = Student()
    ..name = 'Mari'
    ..teacher.value = teacher2;

  await isar.writeTxn(() async => await isar.students.putAll([st, st2, st3]));

  var john = await isar.students.filter().nameEqualTo('John').findFirst();
  await john!.teacher.load();

  var stds = await isar.students
      .filter()
      .teacher((q) => q.subjectEqualTo('Math'))
      .findAll();

  print('Class of ${john.name}`s teacher: ${john.teacher.value!.subject}');
  print("Founded students with `Math` class: ${stds.length}");
```

**Version**
 - Platform: Windows
 - Flutter version: 3.1.0-9.0.pre
 - Isar version: 3.0.0-dev.1 | 2.5.0
 - Isar Generator version: 3.0.0-dev.1 | 2.5.0

**Comment by Viper-Bit** [comment]
@leisim i think this is dup of #390, this is critical bug, when we can expect the new version ?

**Comment by leisim** [comment]
It will probably still take a little time. I can't commit to a date yet unfortunately.

**Comment by leisim** [comment]
Fixed in `3.0.0-dev.8`
